### PR TITLE
Remove wireguard-tools from firezone.

### PR DESCRIPTION
**Pull request by akrousset**
Fixes firezone/product#369

## Summary
I've removed `wireguard-tools.rb` which builds `wg` and other utilities in firezone. 
Any reference to `wg` such as environment variables or config attributes were removed
as well. I also removed private key file handling, because we are generating the private
key using `Wireguardex` and I am setting the public key for the interface in the app env
of `fz_vpn`.

I've also refactored `FzVpn.Interface` to remove the `ip` shellouts during interface creation
as we're still handling that via a shell script that runs from `firezone-ctl`. I've also added a new 
function `list_names` to `FzVpn.Interface` so I can check if the interface is missing and create it 
when needed.

### CODE REVIEW

File: apps/fz_vpn/lib/fz_vpn/interface.ex [hunk 1]
```diff
@@ -18,9 +17,9 @@ defmodule FzVpn.Interface do
   If successful `{:ok, {private_key, public_key}}` is returned. If the interface
   creation fails, `{:error, error_info}` will be logged and returned.
   """
-  def create(name, listen_port, ipv4_address, ipv6_address, mtu) do
+  def create(name, listen_port) do
     private_key = Wireguardex.generate_private_key()
```
Comment (jamilbk) [inline apps/fz_vpn/lib/fz_vpn/interface.ex]: Should this return the `{:ok, private_key}` tuple as well? Alternatively could add exclamation to the function name if you want it to either success or error and crash.

File: apps/fz_vpn/lib/fz_vpn/interface/wg_adapter/live.ex [hunk 2]
```diff
@@ -17,6 +17,10 @@ defmodule FzVpn.Interface.WGAdapter.Live do
     Wireguardex.get_device(name)
   end
 
+  def list_devices do
+    Wireguardex.list_devices()
+  end
+
   def set_device(config, name) do
     Wireguardex.set_device(config, name)
   end
```
Comment (jamilbk) [inline apps/fz_vpn/lib/fz_vpn/interface/wg_adapter/live.ex]:
```suggestion
defdelegate get_device(name), to: Wireguardex
defdelegate list_devices, to: Wireguardex
defdelegate set_device(config, name), to: Wireguardex
```

File: apps/fz_vpn/lib/fz_vpn/interface/wg_adapter/sandbox.ex [hunk 3]
```diff
@@ -35,11 +39,17 @@ defmodule FzVpn.Interface.WGAdapter.Sandbox do
...
+  def handle_call({:list_devices}, _from, devices) do
+    {:reply, {:ok, Map.keys(devices)}, devices}
+  end
+
   @impl GenServer
   def handle_call({:set_device, config, name}, _from, devices) do
     public_key =
       if config.private_key do
-        Wireguardex.get_public_key(config.private_key)
+        {:ok, public_key} = Wireguardex.get_public_key(config.private_key)
```
Comment (jamilbk) [inline apps/fz_vpn/lib/fz_vpn/interface/wg_adapter/sandbox.ex:51]: Remove the conditional and crash if no private key?

```suggestion
{:ok, public_key} = Wireguardex.get_public_key(config.private_key)        
```

**Review by jamilbk — CHANGES_REQUESTED** [review]
This looks good, but I just thought of one thing -- where is the private key persisted?

File: apps/fz_vpn/lib/fz_vpn/interface/wg_adapter/sandbox.ex (hunk 3 above)
Comment (akrousset) [reply apps/fz_vpn/lib/fz_vpn/interface/wg_adapter/sandbox.ex:51]: The private key can be optionally set on the interface with Wireguardex, so I was just mirroring that here.

**Review by jamilbk — DISMISSED** [review]
LGTM. Also added a couple very minor improvements.

**Pull request merged**